gbus/worker.go

```diff
@@ -303,35 +333,46 @@ func (worker *worker) processMessage(delivery amqp.Delivery, isRPCreply bool) {
 				if ackErr != nil {
 					// if this fails then the message will be eventually redeilvered by RabbitMQ
 					//sp handlers should be idempotent.
-					worker.log().WithField("ack_error", ackErr).Warn("failed to send ack to the broker")
+					span.LogFields(slog.Error(ackErr))
+					worker.log().WithError(ackErr).Warn("failed to send ack to the broker")
 				}
 
 			} else {
+				span.LogFields(slog.Error(commitErr))
 				worker.log().WithError(commitErr).Error("failed committing transaction")
 				//if the commit failed we will reject the message
-				worker.Reject(false, delivery)
+				err = worker.Reject(false, delivery)
```

Comment:
It's probably worth invoking it with SafeWithRetries

Reply:
worker.Reject already implements safe retries https://github.com/wework/grabbit/blob/fix-opentracing/gbus/worker.go#L192